### PR TITLE
SynthesizedReadOnlyListTypeSymbol.Create - Check for MissingMetadataTypeSymbol

### PR DESCRIPTION
**Pull request by alrz**
Fix https://github.com/dotnet/roslyn/issues/71297

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Symbols/Synthesized/ReadOnlyListType/SynthesizedReadOnlyListTypeSymbol.cs [hunk 1]
```diff
@@ -102,8 +102,8 @@ internal static NamedTypeSymbol Create(SourceModuleSymbol containingModule, stri
             DiagnosticInfo? diagnosticInfo = null;
 
             var hasReadOnlyInterfaces =
-                !compilation.IsTypeMissing(SpecialType.System_Collections_Generic_IReadOnlyCollection_T)
-                && !compilation.IsTypeMissing(SpecialType.System_Collections_Generic_IReadOnlyList_T);
+                compilation.GetSpecialType(SpecialType.System_Collections_Generic_IReadOnlyCollection_T) is not MissingMetadataTypeSymbol &&
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Symbols/Synthesized/ReadOnlyListType/SynthesizedReadOnlyListTypeSymbol.cs:105]: >compilation.GetSpecialType(SpecialType.System_Collections_Generic_IReadOnlyCollection_T) is not MissingMetadataTypeSymbol && [](http://example.com/codeflow?start=16&length=124)

It looks like we are fixing a real bug, but not adding any unit-tests. I think we should add a unit-test that makes the problem observable. I.e. a test that fails without the fix.

**Comment by alrz** [comment]
@AlekseyTs @dotnet/roslyn-compiler this is ready for review. thanks.

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 4)

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler For the second review

**Review by cston — APPROVED** [review]

**Pull request merged**